list create fails for DeploymentList

@jimmidyson @iocanel 

KubernetesList list = client.inNamespace(namespace).lists().load(is).get();
client.inNamespace(namespace).lists().create(list);

where the inputStream has:
$ cat nginx-deployment.yaml 
```
apiVersion: extensions/v1beta1 
kind: DeploymentList
items:
- apiVersion: extensions/v1beta1
  kind: Deployment
  metadata:
    name: nginx-deployment
    annotations: 
      k8s.starfleet.io/version: "1"
  spec:
    replicas: 3
    revisionHistoryLimit: 1
    template:
      metadata:
        labels:
          app: nginx
      spec:
        containers:
        - name: nginx
          image: nginx:1.7.9
          ports:
          - containerPort: 80
```
results in:
Caused by: java.lang.IllegalStateException: Could not find handler
        at io.fabric8.kubernetes.client.dsl.internal.KubernetesListOperationsImpl.create(KubernetesListOperationsImpl.java:117) ~[kubernetes-client-1.3.83.jar:na]
        at io.fabric8.kubernetes.client.dsl.internal.KubernetesListOperationsImpl.create(KubernetesListOperationsImpl.java:78) ~[kubernetes-client-1.3.83.jar:na]
        at io.fabric8.kubernetes.client.dsl.internal.KubernetesListOperationsImpl.create(KubernetesListOperationsImpl.java:43) ~[kubernetes-client-1.3.83.jar:na]

--- a/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/handlers/ConfigMapHandler.java
+++ b/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/handlers/ConfigMapHandler.java
@@ -1,0 +1,62 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes.client.handlers;
+
+import com.squareup.okhttp.OkHttpClient;
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.ConfigMapBuilder;
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.ResourceHandler;
+import io.fabric8.kubernetes.client.dsl.internal.ConfigMapOperationsImpl;
+import org.apache.felix.scr.annotations.Component;
+import org.apache.felix.scr.annotations.Service;
+
+import java.util.TreeMap;
+
+@Component
+@Service
+public class ConfigMapHandler implements ResourceHandler<ConfigMap, ConfigMapBuilder> {
+
+  @Override
+  public String getKind() {
+    return ConfigMap.class.getSimpleName();
+  }
+
+  @Override
+  public ConfigMap create(OkHttpClient client, Config config, String namespace, ConfigMap item) {
+    return new ConfigMapOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).create();
+  }
+
+  @Override
+  public ConfigMap replace(OkHttpClient client, Config config, String namespace, ConfigMap item) {
+    return new ConfigMapOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).replace(item);
+  }
+
+  @Override
+  public ConfigMap reload(OkHttpClient client, Config config, String namespace, ConfigMap item) {
+    return new ConfigMapOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).fromServer().get();
+  }
+
+  @Override
+  public ConfigMapBuilder edit(ConfigMap item) {
+    return new ConfigMapBuilder(item);
+  }
+
+  @Override
+  public Boolean delete(OkHttpClient client, Config config, String namespace, ConfigMap item) {
+    return new ConfigMapOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).delete(item);
+  }
+}

--- a/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/handlers/DaemonSetHandler.java
+++ b/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/handlers/DaemonSetHandler.java
@@ -1,0 +1,62 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes.client.handlers;
+
+import com.squareup.okhttp.OkHttpClient;
+import io.fabric8.kubernetes.api.model.extensions.DaemonSet;
+import io.fabric8.kubernetes.api.model.extensions.DaemonSetBuilder;
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.ResourceHandler;
+import io.fabric8.kubernetes.client.dsl.internal.DaemonSetOperationsImpl;
+import org.apache.felix.scr.annotations.Component;
+import org.apache.felix.scr.annotations.Service;
+
+import java.util.TreeMap;
+
+@Component
+@Service
+public class DaemonSetHandler implements ResourceHandler<DaemonSet, DaemonSetBuilder> {
+
+  @Override
+  public String getKind() {
+    return DaemonSet.class.getSimpleName();
+  }
+
+  @Override
+  public DaemonSet create(OkHttpClient client, Config config, String namespace, DaemonSet item) {
+    return new DaemonSetOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).create();
+  }
+
+  @Override
+  public DaemonSet replace(OkHttpClient client, Config config, String namespace, DaemonSet item) {
+    return new DaemonSetOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).replace(item);
+  }
+
+  @Override
+  public DaemonSet reload(OkHttpClient client, Config config, String namespace, DaemonSet item) {
+    return new DaemonSetOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).fromServer().get();
+  }
+
+  @Override
+  public DaemonSetBuilder edit(DaemonSet item) {
+    return new DaemonSetBuilder(item);
+  }
+
+  @Override
+  public Boolean delete(OkHttpClient client, Config config, String namespace, DaemonSet item) {
+    return new DaemonSetOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).delete(item);
+  }
+}

--- a/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/handlers/DeploymentHandler.java
+++ b/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/handlers/DeploymentHandler.java
@@ -1,0 +1,62 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes.client.handlers;
+
+import com.squareup.okhttp.OkHttpClient;
+import io.fabric8.kubernetes.api.model.extensions.Deployment;
+import io.fabric8.kubernetes.api.model.extensions.DeploymentBuilder;
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.ResourceHandler;
+import io.fabric8.kubernetes.client.dsl.internal.DeploymentOperationsImpl;
+import org.apache.felix.scr.annotations.Component;
+import org.apache.felix.scr.annotations.Service;
+
+import java.util.TreeMap;
+
+@Component
+@Service
+public class DeploymentHandler implements ResourceHandler<Deployment, DeploymentBuilder> {
+
+  @Override
+  public String getKind() {
+    return Deployment.class.getSimpleName();
+  }
+
+  @Override
+  public Deployment create(OkHttpClient client, Config config, String namespace, Deployment item) {
+    return new DeploymentOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).create();
+  }
+
+  @Override
+  public Deployment replace(OkHttpClient client, Config config, String namespace, Deployment item) {
+    return new DeploymentOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).replace(item);
+  }
+
+  @Override
+  public Deployment reload(OkHttpClient client, Config config, String namespace, Deployment item) {
+    return new DeploymentOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).fromServer().get();
+  }
+
+  @Override
+  public DeploymentBuilder edit(Deployment item) {
+    return new DeploymentBuilder(item);
+  }
+
+  @Override
+  public Boolean delete(OkHttpClient client, Config config, String namespace, Deployment item) {
+    return new DeploymentOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).delete(item);
+  }
+}

--- a/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/handlers/HorizontalPodAutoscalerHandler.java
+++ b/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/handlers/HorizontalPodAutoscalerHandler.java
@@ -1,0 +1,62 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes.client.handlers;
+
+import com.squareup.okhttp.OkHttpClient;
+import io.fabric8.kubernetes.api.model.extensions.HorizontalPodAutoscaler;
+import io.fabric8.kubernetes.api.model.extensions.HorizontalPodAutoscalerBuilder;
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.ResourceHandler;
+import io.fabric8.kubernetes.client.dsl.internal.HorizontalPodAutoscalerOperationsImpl;
+import org.apache.felix.scr.annotations.Component;
+import org.apache.felix.scr.annotations.Service;
+
+import java.util.TreeMap;
+
+@Component
+@Service
+public class HorizontalPodAutoscalerHandler implements ResourceHandler<HorizontalPodAutoscaler, HorizontalPodAutoscalerBuilder> {
+
+  @Override
+  public String getKind() {
+    return HorizontalPodAutoscaler.class.getSimpleName();
+  }
+
+  @Override
+  public HorizontalPodAutoscaler create(OkHttpClient client, Config config, String namespace, HorizontalPodAutoscaler item) {
+    return new HorizontalPodAutoscalerOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).create();
+  }
+
+  @Override
+  public HorizontalPodAutoscaler replace(OkHttpClient client, Config config, String namespace, HorizontalPodAutoscaler item) {
+    return new HorizontalPodAutoscalerOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).replace(item);
+  }
+
+  @Override
+  public HorizontalPodAutoscaler reload(OkHttpClient client, Config config, String namespace, HorizontalPodAutoscaler item) {
+    return new HorizontalPodAutoscalerOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).fromServer().get();
+  }
+
+  @Override
+  public HorizontalPodAutoscalerBuilder edit(HorizontalPodAutoscaler item) {
+    return new HorizontalPodAutoscalerBuilder(item);
+  }
+
+  @Override
+  public Boolean delete(OkHttpClient client, Config config, String namespace, HorizontalPodAutoscaler item) {
+    return new HorizontalPodAutoscalerOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).delete(item);
+  }
+}

--- a/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/handlers/IngressHandler.java
+++ b/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/handlers/IngressHandler.java
@@ -1,0 +1,62 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes.client.handlers;
+
+import com.squareup.okhttp.OkHttpClient;
+import io.fabric8.kubernetes.api.model.extensions.Ingress;
+import io.fabric8.kubernetes.api.model.extensions.IngressBuilder;
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.ResourceHandler;
+import io.fabric8.kubernetes.client.dsl.internal.IngressOperationsImpl;
+import org.apache.felix.scr.annotations.Component;
+import org.apache.felix.scr.annotations.Service;
+
+import java.util.TreeMap;
+
+@Component
+@Service
+public class IngressHandler implements ResourceHandler<Ingress, IngressBuilder> {
+
+  @Override
+  public String getKind() {
+    return Ingress.class.getSimpleName();
+  }
+
+  @Override
+  public Ingress create(OkHttpClient client, Config config, String namespace, Ingress item) {
+    return new IngressOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).create();
+  }
+
+  @Override
+  public Ingress replace(OkHttpClient client, Config config, String namespace, Ingress item) {
+    return new IngressOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).replace(item);
+  }
+
+  @Override
+  public Ingress reload(OkHttpClient client, Config config, String namespace, Ingress item) {
+    return new IngressOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).fromServer().get();
+  }
+
+  @Override
+  public IngressBuilder edit(Ingress item) {
+    return new IngressBuilder(item);
+  }
+
+  @Override
+  public Boolean delete(OkHttpClient client, Config config, String namespace, Ingress item) {
+    return new IngressOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).delete(item);
+  }
+}

--- a/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/handlers/JobHandler.java
+++ b/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/handlers/JobHandler.java
@@ -1,0 +1,62 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes.client.handlers;
+
+import com.squareup.okhttp.OkHttpClient;
+import io.fabric8.kubernetes.api.model.extensions.Job;
+import io.fabric8.kubernetes.api.model.extensions.JobBuilder;
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.ResourceHandler;
+import io.fabric8.kubernetes.client.dsl.internal.JobOperationsImpl;
+import org.apache.felix.scr.annotations.Component;
+import org.apache.felix.scr.annotations.Service;
+
+import java.util.TreeMap;
+
+@Component
+@Service
+public class JobHandler implements ResourceHandler<Job, JobBuilder> {
+
+  @Override
+  public String getKind() {
+    return Job.class.getSimpleName();
+  }
+
+  @Override
+  public Job create(OkHttpClient client, Config config, String namespace, Job item) {
+    return new JobOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).create();
+  }
+
+  @Override
+  public Job replace(OkHttpClient client, Config config, String namespace, Job item) {
+    return new JobOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).replace(item);
+  }
+
+  @Override
+  public Job reload(OkHttpClient client, Config config, String namespace, Job item) {
+    return new JobOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).fromServer().get();
+  }
+
+  @Override
+  public JobBuilder edit(Job item) {
+    return new JobBuilder(item);
+  }
+
+  @Override
+  public Boolean delete(OkHttpClient client, Config config, String namespace, Job item) {
+    return new JobOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).delete(item);
+  }
+}

--- a/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/handlers/ReplicaSetHandler.java
+++ b/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/handlers/ReplicaSetHandler.java
@@ -1,0 +1,63 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes.client.handlers;
+
+import com.squareup.okhttp.OkHttpClient;
+import io.fabric8.kubernetes.api.model.extensions.ReplicaSet;
+import io.fabric8.kubernetes.api.model.extensions.ReplicaSetBuilder;
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.ResourceHandler;
+import io.fabric8.kubernetes.client.dsl.internal.ReplicaSetOperationsImpl;
+import org.apache.felix.scr.annotations.Component;
+import org.apache.felix.scr.annotations.Service;
+
+import java.util.TreeMap;
+import java.util.concurrent.TimeUnit;
+
+@Component
+@Service
+public class ReplicaSetHandler implements ResourceHandler<ReplicaSet, ReplicaSetBuilder> {
+
+  @Override
+  public String getKind() {
+    return ReplicaSet.class.getSimpleName();
+  }
+
+  @Override
+  public ReplicaSet create(OkHttpClient client, Config config, String namespace, ReplicaSet item) {
+    return new ReplicaSetOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>(), false, config.getRollingTimeout(), TimeUnit.MILLISECONDS).create();
+  }
+
+  @Override
+  public ReplicaSet replace(OkHttpClient client, Config config, String namespace, ReplicaSet item) {
+    return new ReplicaSetOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>(), false, config.getRollingTimeout(), TimeUnit.MILLISECONDS).replace(item);
+  }
+
+  @Override
+  public ReplicaSet reload(OkHttpClient client, Config config, String namespace, ReplicaSet item) {
+    return new ReplicaSetOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>(), false, config.getRollingTimeout(), TimeUnit.MILLISECONDS).fromServer().get();
+  }
+
+  @Override
+  public ReplicaSetBuilder edit(ReplicaSet item) {
+    return new ReplicaSetBuilder(item);
+  }
+
+  @Override
+  public Boolean delete(OkHttpClient client, Config config, String namespace, ReplicaSet item) {
+    return new ReplicaSetOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>(), false, config.getRollingTimeout(), TimeUnit.MILLISECONDS).delete(item);
+  }
+}

--- a/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/handlers/ReplicationControllerHandler.java
+++ b/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/handlers/ReplicationControllerHandler.java
@@ -35,6 +35,7 @@ public class ReplicationControllerHandler implements ResourceHandler<Replication
   public String getKind() {
     return ReplicationController.class.getSimpleName();
   }
+
   @Override
   public ReplicationController create(OkHttpClient client, Config config, String namespace, ReplicationController item) {
     return new ReplicationControllerOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>(), false, config.getRollingTimeout(), TimeUnit.MILLISECONDS).create();

--- a/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/handlers/ThirdPartyResourceHandler.java
+++ b/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/handlers/ThirdPartyResourceHandler.java
@@ -1,0 +1,62 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes.client.handlers;
+
+import com.squareup.okhttp.OkHttpClient;
+import io.fabric8.kubernetes.api.model.extensions.ThirdPartyResource;
+import io.fabric8.kubernetes.api.model.extensions.ThirdPartyResourceBuilder;
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.ResourceHandler;
+import io.fabric8.kubernetes.client.dsl.internal.ThirdPartyResourceOperationsImpl;
+import org.apache.felix.scr.annotations.Component;
+import org.apache.felix.scr.annotations.Service;
+
+import java.util.TreeMap;
+
+@Component
+@Service
+public class ThirdPartyResourceHandler implements ResourceHandler<ThirdPartyResource, ThirdPartyResourceBuilder> {
+
+  @Override
+  public String getKind() {
+    return ThirdPartyResource.class.getSimpleName();
+  }
+
+  @Override
+  public ThirdPartyResource create(OkHttpClient client, Config config, String namespace, ThirdPartyResource item) {
+    return new ThirdPartyResourceOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).create();
+  }
+
+  @Override
+  public ThirdPartyResource replace(OkHttpClient client, Config config, String namespace, ThirdPartyResource item) {
+    return new ThirdPartyResourceOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).replace(item);
+  }
+
+  @Override
+  public ThirdPartyResource reload(OkHttpClient client, Config config, String namespace, ThirdPartyResource item) {
+    return new ThirdPartyResourceOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).fromServer().get();
+  }
+
+  @Override
+  public ThirdPartyResourceBuilder edit(ThirdPartyResource item) {
+    return new ThirdPartyResourceBuilder(item);
+  }
+
+  @Override
+  public Boolean delete(OkHttpClient client, Config config, String namespace, ThirdPartyResource item) {
+    return new ThirdPartyResourceOperationsImpl(client, config, null, namespace, null, true, item, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>()).delete(item);
+  }
+}

--- a/kubernetes-client/src/main/resources/META-INF/services/io.fabric8.kubernetes.client.ResourceHandler
+++ b/kubernetes-client/src/main/resources/META-INF/services/io.fabric8.kubernetes.client.ResourceHandler
@@ -14,16 +14,24 @@
 # limitations under the License.
 #
 
+io.fabric8.kubernetes.client.handlers.ConfigMapHandler
+io.fabric8.kubernetes.client.handlers.DaemonSetHandler
+io.fabric8.kubernetes.client.handlers.DeploymentHandler
 io.fabric8.kubernetes.client.handlers.EndpointsHandler
 io.fabric8.kubernetes.client.handlers.EventHandler
+io.fabric8.kubernetes.client.handlers.HorizontalPodAutoscalerHandler
+io.fabric8.kubernetes.client.handlers.IngressHandler
+io.fabric8.kubernetes.client.handlers.JobHandler
 io.fabric8.kubernetes.client.handlers.NamespaceHandler
 io.fabric8.kubernetes.client.handlers.NodeHandler
 io.fabric8.kubernetes.client.handlers.PersistentVolumeClaimHandler
 io.fabric8.kubernetes.client.handlers.PersistentVolumeHandler
 io.fabric8.kubernetes.client.handlers.PodHandler
+io.fabric8.kubernetes.client.handlers.ReplicaSetHandler
 io.fabric8.kubernetes.client.handlers.ReplicationControllerHandler
 io.fabric8.kubernetes.client.handlers.ResourceQuotaHandler
 io.fabric8.kubernetes.client.handlers.SecretHandler
 io.fabric8.kubernetes.client.handlers.SecurityContextConstraintsHandler
 io.fabric8.kubernetes.client.handlers.ServiceAccountHandler
 io.fabric8.kubernetes.client.handlers.ServiceHandler
+io.fabric8.kubernetes.client.handlers.ThirdPartyResourceHandler

